src/main/java/hudson/remoting/Engine.java

```diff
@@ -1034,16 +1036,18 @@ private static FileInputStream getFileInputStream(final File file) throws Privil
         });
     }
 
-    private SSLSocketFactory getSSLSocketFactory()
+    @CheckForNull
+    @Restricted(NoExternalUse.class)
+    static SSLSocketFactory getSSLSocketFactory(List<X509Certificate> x509Certificates)
```

Comment:
Widening the visibility of this method and making static so that it can be reused in another place where this same logic is needed. Also added a `@CheckForNull` annotation while I was here, since this method can sometimes return null.